irohad/model/converters/impl/json_query_factory.cpp

```diff
@@ -15,241 +15,106 @@
-                            return not obj_query.HasMember(field);
-                          })) {
-            log_->error("No required fields in json");
-            return nonstd::nullopt;
-          }
-        }
-
-        // check if all member values have valid type.
-        {
-          if (not obj_query["signature"].IsObject()) {
-            log_->error("Type mismatch in json,  signature must be object");
-            return nonstd::nullopt;
-          }
-
-          if (not obj_query["creator_account_id"].IsString()) {
-            log_->error("Type mismatch in json,  created_js must be string");
-            return nonstd::nullopt;
-          }
-
-          if (not obj_query["created_ts"].IsUint64()) {
-            log_->error("Type mismatch in json,  created_ts must be uint64");
-            return nonstd::nullopt;
-          }
-
-          if (not obj_query["query_counter"].IsUint64()) {
-            log_->error("Type mismatch in json,  query_counter must be uint64");
-            return nonstd::nullopt;
-          }
-
-          if (not obj_query["query_type"].IsString()) {
-            log_->error("Type mismatch in json,  query_type must be string");
-            return nonstd::nullopt;
-          }
-        }
-
-        auto sig = obj_query["signature"].GetObject();
-
-        // check if signature has all needed fields
-        {
-          if (not sig.HasMember("pubkey")) {
-            log_->error("No pubkey in signature in json");
-            return nonstd::nullopt;
-          }
-          if (not sig.HasMember("signature")) {
-            log_->error("No signature in json");
-            return nonstd::nullopt;
-          }
-        }
-
-        // check if members of signature have valid type.
-        {
-          if (not sig["pubkey"].IsString()) {
-            log_->error(
-                "Type mismatch in json,  pubkey in signature must be string");
-            return nonstd::nullopt;
-          }
-          if (not sig["signature"].IsString()) {
-            log_->error(
-                "Type mismatch in json,  signature in signature must be string");
-            return nonstd::nullopt;
-          }
-        }
-
-        auto query_type = obj_query["query_type"].GetString();
-        auto it = deserializers_.find(query_type);
-        if (it != deserializers_.end()) {
-          auto query = (this->*it->second)(obj_query);
-          if (not query) {
-            return nonstd::nullopt;
-          }
-
-          // Set query signature
-          Signature signature;
-          hexstringToArray(sig["pubkey"].GetString(), signature.pubkey);
-          hexstringToArray(sig["signature"].GetString(), signature.signature);
-          query->signature = signature;
-          return query;
-        }
-        log_->error("No query type found");
-        return nonstd::nullopt;
+        return stringToJson(query_json)
+            | [this](auto &json) { return this->deserialize(json); };
```

Comment:
`this` can be omitted afair

Reply:
`error: cannot call member function ‘iroha::model::optional_ptr<iroha::model::Query> iroha::model::converters::JsonQueryFactory::deserialize(const Document&)’ without object`

Reply:
[Weird](https://ideone.com/ZVg648), but anyway not a problem